src/platform/assets/composables/useAssetBrowserDialog.ts

```diff
@@ -17,30 +19,54 @@ interface AssetBrowserDialogProps {
   onAssetSelected?: (filename: string) => void
 }
 
+interface BrowseOptions {
+  /** Asset type tag to filter by (e.g., 'models') */
+  assetType: string
+  /** Custom modal title (optional) */
+  title?: string
+  /** Called when asset selected */
+  onAssetSelected?: (asset: AssetItem) => void
```

Comment:
How does this differ from the other onAssetSelected?

Reply:
now that i think about it, i should just make them both an AssetItem